### PR TITLE
[SPARK-8085] [SPARKR] Support user-specified schema in read.df

### PR DESCRIPTION
**Pull request by shivaram**
cc @davies @sun-rui

### CODE REVIEW

File: R/pkg/R/SQLContext.R [hunk 1]
```diff
@@ -462,14 +462,19 @@ read.df <- function(sqlContext, path = NULL, source = NULL, ...) {
     source <- callJMethod(sqlContext, "getConf", "spark.sql.sources.default",
                           "org.apache.spark.sql.parquet")
   }
-  sdf <- callJMethod(sqlContext, "load", source, options)
```
Comment (davies) [inline R/pkg/R/SQLContext.R]: The load() in Scala is deprecated in 1.4, use sqlContext.read.load() instead.

Comment (shivaram) [inline R/pkg/R/SQLContext.R]: Its actually harder to use the chaining API from SparkR -- We'll need to get a jobj for sqlContext.read and then set schema on it and then call load on it. Not sure its worth to make 3 RPCs from R to JVM ?

Comment (davies) [inline R/pkg/R/SQLContext.R]: But the deprecated one will generate a warning, it's not good.

cc @rxin

Comment (rxin) [inline R/pkg/R/SQLContext.R]: how about creating some R helper functions and just call that once?

Comment (shivaram) [inline R/pkg/R/SQLContext.R]: Hmm - yes. I could add something to `SQLUtils` in `r/api` I guess. But in general why did we need to deprecate the `load()` functions ?

Comment (rxin) [inline R/pkg/R/SQLContext.R]: They are growing to over 30+ various variants of input/output functions. We now dump all of them in a namespace.

Comment (shivaram) [inline R/pkg/R/SQLContext.R]: Alright I use a wrapper function `loadDF` in SQLContext now

File: R/pkg/R/SQLContext.R [hunk 2]
```diff
@@ -462,14 +462,20 @@ read.df <- function(sqlContext, path = NULL, source = NULL, ...) {
...
+  }
   dataFrame(sdf)
 }
 
 #' @aliases loadDF
 #' @export
 
-loadDF <- function(sqlContext, path = NULL, source = NULL, ...) {
+loadDF <- function(sqlContext, path = NULL, source = NULL, schema = NULL, ...) {
   read.df(sqlContext, path, source, ...)
```
Comment (davies) [inline R/pkg/R/SQLContext.R]: We should pass schema into read.df.

**Comment by shivaram** [comment]
Jenkins, retest this please

File: R/pkg/R/SQLContext.R (hunk 2 above)
Comment (shivaram) [inline R/pkg/R/SQLContext.R]: Good catch. Fixed now and also added a unit test for loadDF

**Comment by shivaram** [comment]
Jenkins, retest this please

**Comment by davies** [comment]
LGTM, thanks!

**Pull request closed without merging**